src/datasets/load.py

```diff
@@ -1608,7 +1608,10 @@ def _get_modification_time(module_hash):
             }
             return DatasetModule(module_path, hash, builder_kwargs, importable_file_path=importable_file_path)
         cache_dir = os.path.expanduser(str(self.cache_dir or config.HF_DATASETS_CACHE))
-        cached_datasets_directory_path_root = os.path.join(cache_dir, self.name.replace("/", "___"))
+        dataset_name = Path(self.name).name
+        dataset_name_snakecase = camelcase_to_snakecase(dataset_name)
+        cached_relative_path = self.name.replace("/", "___").replace(dataset_name, dataset_name_snakecase)
```

Comment:
This can lead to issues if the dataset name is also in the namespace, here is a suggestion:

```suggestion
        namespace_and_dataset_name = self.name.split("/")
        namespace_and_dataset_name[-1] = camelcase_to_snakecase(namespace_and_dataset_name[-1])
        cached_relative_path = "___".join(dataset_name_parts)
```